### PR TITLE
cmake // Added Ctest support for unittests

### PR DESCRIPTION
**Pull request by snikulov**
Hi all,
I've enabled ctest verification for cmake builds and ```tests/unit``` folder.
Now it is possible after build run
on Linux
```make test```
on Windows
```ctest -VV -C <Debug|Release>```

### CODE REVIEW

**Comment by snikulov** [comment]
@bradking Brad, could you please review and comment? Thank you!

File: CMakeLists.txt [hunk 1]
```diff
@@ -1042,6 +1042,8 @@ if(BUILD_CURL_EXE)
   add_subdirectory(src)
 endif()
 if(BUILD_CURL_TESTS)
+  include(CTest)
+  enable_testing()
```
Comment (bradking) [inline CMakeLists.txt]: The `CTest` module calls `enable_testing` for us so we don't need to call it here.

File: tests/unit/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+foreach(_testfile ${UT_SRC})
+
+  get_filename_component(_testname ${_testfile} NAME_WE)
+  add_executable(${_testname} ${_testfile} ${UT_COMMON_FILES})
+  target_link_libraries(${_testname} libcurl ${CURL_LIBS})
+  set_target_properties(${_testname}
+      PROPERTIES COMPILE_DEFINITIONS "UNITTESTS")
+
+  add_test(NAME ${_testname}
+           WORKING_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
```
Comment (bradking) [inline tests/unit/CMakeLists.txt]: I don't think `EXECUTABLE_OUTPUT_PATH` is set anywhere.  I think this line can just be dropped.

File: CMakeLists.txt (hunk 1 above)
Comment (snikulov) [inline CMakeLists.txt]: @bradking From which version? Looks I'm a bit outdated here.

File: tests/unit/CMakeLists.txt (hunk 2 above)
Comment (snikulov) [inline tests/unit/CMakeLists.txt]: @bradking Agreed. It was copy-paste from my project. But anyway, sometimes it is required for DLL when shared library used in build.

File: CMakeLists.txt (hunk 1 above)
Comment (bradking) [inline CMakeLists.txt]: It has always done this.

Comment (bradking) [inline CMakeLists.txt]: Note that it depends on `BUILD_TESTING`.

**Comment by snikulov** [comment]
@bradking updated and rebased into single commit.

**Comment by bradking** [comment]
The `CTest` module defines a `BUILD_TESTING` option that controls most of its functionality, including its call to `enable_testing`.  The intended way to use the `CTest` module is to include it unconditionally and then honor its `BUILD_TESTING` option:

```cmake
include(CTest)
if(BUILD_TESTING)
  add_subdirectory(tests)
endif()
```

However, curl is currently defining its own BUILD_CURL_TESTS option.  It also has a code path that includes the `CTest` module earlier in a `BUILD_DASHBOARD_REPORTS` condition.  These should all be reconciled.

**Comment by snikulov** [comment]
@bradking So what is your proposal, exactly? Should I remove BUILD_CURL_TESTS and BUILD_DASHBOARD_REPORTS anywhere in scripts?

**Comment by bradking** [comment]
> Should I remove BUILD_CURL_TESTS and BUILD_DASHBOARD_REPORTS anywhere in scripts?

That would be the simplest solution.  Since the CMake files still warn that they are poorly maintained I don't think anyone expects stability in the current options.

**Comment by snikulov** [comment]
@bradking updated.

**Comment by bradking** [comment]
The change itself looks good.  Please squash in the fixups and revise the commit message to explain the new logic change.

**Comment by snikulov** [comment]
@bradking Done.

@bagder Daniel, Could you please merge?

**Comment by bagder** [comment]
Done, thanks!

Small nit: please make the commit message to follow our commit styles and I'll have an easier job merging your work!

**Pull request closed without merging**